test/environment_variable.c

```diff
@@ -0,0 +1,169 @@
+
+/*
+ * Copyright (c) 2016 Jonathan Anderson
+ * All rights reserved.
+ *
+ * This software was developed at Memorial University under the
+ * NSERC Discovery program (RGPIN-2015-06048).
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+/*
+ * RUN: %cc -D PARENT %cflags -D LIBPRELOAD="\"%dir\"lib" -D TEST_DATA_DIR="\"%p/Inputs\"" %s %ldflags -o %t.parent
+ * RUN: %cc -D CHILD %cflags -D TEST_DATA_DIR="\"%p/Inputs\"" %s %ldflags -o %t.child
+ * RUN: %t.parent %t.child > %t.out
+ * RUN: %filecheck %s -input-file %t.out
+ */
+
+#ifdef PARENT
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<unistd.h>
+#include <assert.h>
+#include <unistd.h>
+#include <sys/capsicum.h>
+#include <sys/file.h>
+#include <sys/mman.h>
+#include <string.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include"libpreopen.h"
```

Comment:
If any of these include are common to both the parent and child, you could move them outside of the conditionally-compiled blocks, i.e., before `#ifdef PARENT`.